docs/qa/docker.rst

```diff
@@ -0,0 +1,116 @@
+Docker for QA
+=============
+The following describes how to get a prebuilt instance of Normandy for
+testing. This is not suitable for development.
+
+Prerequisites
+-------------
+This guide assumes you have already installed and set up the following:
+
+1. Docker_
+2. `Postgres 9.4`_
+
+.. _Docker: https://docs.docker.com/engine/installation/
+.. _Postgres 9.4: http://www.postgresql.org/
+
+Preparing an environment file
+-----------------------------
+Normandy gets all of its configuration via environment variables. Docker
+has built in support for this pattern. For this set up, we will be using
+and environment file to specifiy this configuration, and having Docker
+read that file.
+
+The file can be named anything, but canonically is named ``.env``. The
+only thing that is required is a database config, which looks something
+like the below. We'll go over how to fill in the blanks next.
+
+.. code-block:: ini
+
+  DATABASE_URL=postgresql://<username>:<password>@<host>/<db_name>
+
+``<username>`` and ``<password>`` are the details to authenticate with
+Postgres. ``<db_name>`` is the name of the database to connect to, and
+is usually ``normandy``.
+
+``<host>`` is more complicated. This needs to be something that the
+Docker container can use to connect to Postgres. There are a few options here:
+
+Linux with local Postgres
+  Use ``172.17.0.1``
+
+_`Docker Toolbox` with local Postgres (OSX or Windows)
+  Use ``192.168.99.1``
+
+.. _the Docker Toolbox: https://docs.docker.com/engine/installation/mac/
+
+Configuring Postgres
+--------------------
+
+1. Listening on the Docker IP:
+
+  By default Postgres only listens on localhost, and so Docker containers
+  won't be able to connect to it. To change this, edit your Postgres config
+  (it is in the Postgres data directory, often at
+  ``/var/lib/postgres/postgresql.conf`` or similar). There should be a commented
+  out line like:
+
+  .. code-block:: ini
+
+    #listen_addresses = 'localhost'
+
+  Uncomment it, and change it to include the host IP address you found above,
+  like:
+
+  .. code-block:: ini
+
+    listen_addresses = 'localhost,172.17.0.1'
+
+2. Allowing connections from the Docker subnet:
+
+   Postgres has ACLs per IP, so we'll need to allow connections from the Docker
+   network. Edit your Postgres HBA file, ``pg_hba.conf``. It should be in the
+   same place as ``postgresql.conf``. Add a line referencing the docker
+   network, like this:
+
+   .. code-block:: ini
+
+     host    all             all             172.17.0.0/16           trust
+
+   For Docker Toolbox, use ``192.168.99.0/24``.
+
+Running Normandy
```

Comment:
Can we write a script that does this and outputs the IP, but doesn't run it in the background, so that the person running it can halt it manually? The more like "runserver" this whole process is the better.